compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -541,7 +541,11 @@ class Namer { typer: Typer =>
           res = cpy.TypeDef(modCls)(
             rhs = cpy.Template(modTempl)(
               derived = if (fromTempl.derived.nonEmpty) fromTempl.derived else modTempl.derived,
-              body = fromTempl.body ++ modTempl.body))
+              body = fromTempl.body.filter {
+                  case stat: DefDef => stat.name != nme.toString_
```

Comment:
We should also check the number of parameters to only match `def toString` and `def toString()`, but not `def toString(i: Int)`.

Reply:
No because we are scanning the synthesized companion object. That one can only contain one version of `toString`.